javascript/javascript2/week1/spiritAnimal.js

```diff
@@ -0,0 +1,44 @@
+//  DOM elements
+const nameTag=document.getElementById("name-input");
+const animalBtn=document.getElementById("generate-btn");
+const spiritAnimalElement=document.getElementById("spirit-animal");
+const newAnimalBtn=document.getElementById("new-animal-btn");
+
+// Array of spirit animal names
+
+const arrayspiritAnimals = [
+    "The Grey Wolf",
+    "The Golden Eagle",
+    "The Wise Owl",
+    "The Dancing Peacock",
+    "The Crying Butterfly",
+    "The Roaring Lion",
+    "The Soaring Hawk",
+    "The Strong Serpant",
+    "The Mighty Bear",
+    "The Graceful Deer"
+  ];
+  
+  // Build spirit animal name
+
+  function generateSpiritAnimal(){
+    const name=nameTag.value.trim(); // trim has the role of eliminating white space and cheching is the string is empty or not
+    if(name=== ""){
```

Comment:
Sometimes the value can be null or undefined, due to the nature of JavaScript/HTML. 
So, for such cases, it is safer to write `if(!name)`, which translate to "If name is an empty string, or null or undefined..."
Take a look here: https://developer.mozilla.org/en-US/docs/Glossary/Falsy